django/forms/widgets.py

```diff
@@ -180,11 +180,11 @@ def format_value(self, value):
         """
         Return a value as it should appear when rendered in a template.
         """
-        if value is None:
-            value = ''
+        if value == '':
+            value = None
```

Comment:
`formats.localize_input(None)` doesn't look like it does anything --- would be be fine `return None` here to avoid `if value is not None else value` later?